packages/beacon-node/src/network/processor/gossipHandlers.ts

```diff
@@ -485,6 +556,79 @@ export function getGossipHandlers(modules: ValidatorFnsModules, options: GossipH
   };
 }
 
+/**
+ * For now, only beacon_attestation topic is batched.
+ */
+function getBatchHandlers(modules: ValidatorFnsModules, options: GossipHandlerOpts): Partial<BatchGossipHandlers> {
+  const {chain, metrics, logger, aggregatorTracker} = modules;
+  return {
+    [GossipType.beacon_attestation]: async (
+      gossipHandlerParams: GossipHandlerParamGeneric<GossipType.beacon_attestation>[]
+    ): Promise<(null | AttestationError)[]> => {
+      const results: (null | AttestationError)[] = [];
+      const attestationCount = gossipHandlerParams.length;
+      if (attestationCount === 0) {
+        return results;
+      }
+      // all attestations should have same attestation data as filtered by network processor
+      const {subnet, fork} = gossipHandlerParams[0].topic;
+      const validationParams = gossipHandlerParams.map((param) => ({
+        attestation: null,
+        serializedData: param.gossipData.serializedData,
+        attSlot: param.gossipData.msgSlot,
+        attDataBase64: param.gossipData.indexed,
+      })) as AttestationOrBytes[];
+      const {results: validationResults, batchableBls} = await validateGossipAttestationsSameAttData(
+        fork,
+        chain,
+        validationParams,
+        subnet
+      );
+      for (const [i, validationResult] of validationResults.entries()) {
+        if (validationResult.err) {
+          results.push(validationResult.err as AttestationError);
+          continue;
+        }
+
+        results.push(null);
```

Comment:
You can move this at the bottom with a 
```ts
return validationResults.map(..)
```

Reply:
I don't want to loop through the `validationResults` again, I can add some comments for this `null` value instead